### DISCUSSION TITLE
Questions from a student wanting to build something similiar to mafs

### DISCUSSION THREAD

**Discussion by mattiastofte** [stevenpetryk/mafs#54] [General]
Hi! My name is Matt and I'm currently studying computer sciences at NTNU in Norway. I'm also a part of a student organisation looking to improve education in Norway with a platform that provides courses at higher education. 

We've always talked about creating our own library that would be similar to manim while being embeddable into our webpage. For performance and development reasons we went with typescript as our preferred language for this. This is the reason why we have been researching libraries like mafs, as we need a lot of inspiration so this library can be both performant and very useful for our students.

In an ideal world we would use mafs, although it misses some key features we are really interested in. One of them is animation. We have a lot of explanatory videos and it would be really neet if this library could be used in a custom editor to create such videos. Other features we are interested in are zooming, latex, text-boxes, etc needed to educate our users.

Our question to you is: would you do something different if you made mafs from the ground-up today? Do you have any tips? What is the reason that animation is hard to implement per this date? Any help is greatly appreciated.

**Comment by stevenpetryk** [comment]
Hey Matt,

Animation isn't hard per se, it's just that Mafs isn't my job so I haven't had enough time or energy to work on it consistently.

I wouldn't do anything differently if I build it from the ground up.

If animation is your _primary_ use case, you might get better performance out of something based on WebGL, but I chose SVG for this project for simplicity. It performs well enough.

**Reply by mattiastofte** [reply]
Thanks @stevenpetryk for your valuable feedback:) Mafs is an incredible library and you've done an incredible job with it. We're just in the researching phase and it will be many months before we start with some active development. Maybe we'll just end up forking and or improving mafs so it suits our needs.

Anyways, thanks again and have a wonderful day!
